src/derive/curve.rs

```diff
@@ -132,58 +192,41 @@ macro_rules! new_curve_impl {
                     }
 
                     fn from_bytes_unchecked(bytes: &Self::Repr) -> CtOption<Self> {
+                        // In compressed form we cannot skip the curve check.
                         Self::from_bytes(bytes)
                     }
 
                     fn to_bytes(&self) -> Self::Repr {
                         if bool::from(self.is_identity()) {
                             let mut bytes = [0; [< $name _COMPRESSED_SIZE >]];
-                            bytes[[< $name _COMPRESSED_SIZE >] - 1] |= 0b1000_0000;
+                            // When spare_bits == 1, the identity is all 0's.
+                            if $spare_bits == 0 || $spare_bits ==2 {
+                                bytes[[< $name _FLAG_BYTE_INDEX>]] |= IS_IDENTITY_MASK;
+                            }
                             [< $name Compressed >](bytes)
                         } else {
                             let (x, y) = (self.x, self.y);
-                            let sign = (y.to_bytes()[0] & 1) << 6;
                             let mut xbytes = [0u8; [< $name _COMPRESSED_SIZE >]];
                             xbytes[..$base::size()].copy_from_slice(&x.to_bytes());
```

Comment:
Note that the endianess depends on the `to_bytes()` and `from_bytes()` implementations of the `$base` type.